### ISSUE TITLE
Collision glitch with polygon sliding over simple edges with ghost vertices

### ISSUE THREAD

**Issue by GoogleCodeExporter** [erincatto/box2d#341]
```
See the attached testbed demo. A polygon slides over 3 connected edge shapes 
(using ghost vertices), which form a very simple terrain with no harsh angles. 
When passing from the horizontal edge to the first downsloping edge, there in a 
collision which jerks the polygon back.

Treating the terrain as 4 vertices A, B, C, D with edges ()AB(C), (A)BC(D) and 
(B)CD(), where parentheses represent the previous/next vertices of the edge 
shape:

The unwanted collision occurs when the polygon first reaches vertex B.
If either the previous or next vertices of BC are removed, the collision goes 
away and behaviour is as expected.
Bizarrely, with both ghost vertices in place, whether or not the collision 
occurs is dependent upon the coordinates of vertex D - two vertices away from 
the point of collision.

Tested with Box2d v2.3.1
```

Original issue reported on code.google.com by `cjan...@gmail.com` on 20 Dec 2014 at 8:35

Attachments:
- [collision_issue.png](https://storage.googleapis.com/google-code-attachments/box2d/issue-341/comment-0/collision_issue.png)
- [Skier.h](https://storage.googleapis.com/google-code-attachments/box2d/issue-341/comment-0/Skier.h)

**Comment by louis-langholtz** [comment]
Incidentally, I did some work in [my dev branch fork of Box2D](https://github.com/louis-langholtz/Box2D) to change the way collision responses were generated for polygons sliding over any multi-vertex shape. Was able to get smooth sliding of polygons over polygons without using anything like "ghost" vertices. Also smooth sliding over edge shapes (without ghost vertices).

I have [a write-up of sorts of these changes that I put online](https://github.com/louis-langholtz/Box2D/blob/dev/Documentation/CollisionHandlng.md). It's not perfect but it was an interesting endeavor for me and I'd be happy if it's interesting to anyone else too.

**Comment by erincatto** [comment]
Fixed in #607
